app/src/main/java/de/grobox/transportr/utils/IntentUtils.kt

```diff
@@ -86,22 +96,33 @@ object IntentUtils {
         val geo = Uri.parse(uri1 + uri2)
 
         // show station on external map
-        val intent = Intent(Intent.ACTION_VIEW)
-        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
-        intent.data = geo
-        if (intent.resolveActivity(context.packageManager) != null) {
-            Log.d(context.javaClass.simpleName, "Starting geo intent: " + geo.toString())
-            context.startActivity(intent)
+        val intent = Intent(Intent.ACTION_VIEW).apply {
+            flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
+            data = geo
+        }
+        val intents = context.packageManager.queryIntentActivities(intent, 0).mapNotNull {
+            val packageName = it.activityInfo.packageName
+            if (context.packageName != packageName)
+                Intent(intent).apply { `package` = packageName }
+            else null
```

Comment:
Do we need to set the intent's package name here?

Reply:
Yes, that is how it was done in https://stackoverflow.com/questions/38345994/how-to-exclude-a-specific-application-from-action-send-intent.

Basically, we have multiple intents filtering for one specific `packageName` each to be passed in `Intent.EXTRA_INITIAL_INTENTS`.

For API >= 24 it would have been simpler by just using `Intent.EXTRA_EXCLUDE_COMPONENTS`.